### PR TITLE
Add login and logout command 

### PR DESCRIPTION
**Pull request by rcvalerio**
closes https://github.com/inductiva/tasks/issues/419

This pull request improves the user onboarding experience for both Windows and Linux users by eliminating the need for environment variables when logging in. The API key is now stored in a local file, making it easier for users to set up without having to manually configure environment variables.

### Key Changes:
- The `INDUCTIVA_API_KEY` environment variable is still supported for backward compatibility, but it will only be used if defined. If both the environment variable and the local file are present, the environment variable will take priority.
- Users can now use the `inductiva auth login` command to authenticate and store their API key in a local file within the inductiva folder. `inductiva auth logout` will remove the local file and logout the user.

### Possible improvements:
- Fix existing bug: https://github.com/inductiva/tasks/issues/585;
- Encrypt API key: huggingface and wandb do not encrypt it;

### Important:
- To test this command it is recommended to not have `INDUCTIVA_API_KEY` set, otherwise inductiva will always use the environment variable.

### Tested on:
- [x] Linux;
- [x] Windows;
- [x] Mac

### Example Workflow:

**1. Logging In (First Time):**
```
$ inductiva auth login
     ___  _   _  ____   _   _   ____  _____  ___ __     __ _    
    |_ _|| \ | ||  _ \ | | | | / ___||_   _||_ _|\ \   / // \   
     | | |  \| || | | || | | || |      | |   | |  \ \ / // _ \  
     | | | |\  || |_| || |_| || |___   | |   | |   \ V // ___ \ 
    |___||_| \_||____/  \___/  \____|  |_|  |___|   \_//_/   \_\
    
    To log in, you need an API key. You can obtain it from your account at https://console.inductiva.ai/account.
Please paste your API key here (input will not be visible): 
Welcome back rvalerio!
```

**2. Logged In (Already logged in):**
```
$ inductiva auth login
     ___  _   _  ____   _   _   ____  _____  ___ __     __ _    
    |_ _|| \ | ||  _ \ | | | | / ___||_   _||_ _|\ \   / // \   
     | | |  \| || | | || | | || |      | |   | |  \ \ / // _ \  
     | | | |\  || |_| || |_| || |___   | |   | |   \ V // ___ \ 
    |___||_| \_||____/  \___/  \____|  |_|  |___|   \_//_/   \_\
    
    You are already logged in. Run `inductiva logout` if you want to log out. 
    Setting a new API key will erase the existing one.
    To log in, you need an API key. You can obtain it from your account at https://console.inductiva.ai/account.
Please paste your API key here (input will not be visible): 
Welcome back rvalerio!
```

**2. Invalid api key**
```
$ inductiva auth login
     ___  _   _  ____   _   _   ____  _____  ___ __     __ _    
    |_ _|| \ | ||  _ \ | | | | / ___||_   _||_ _|\ \   / // \   
     | | |  \| || | | || | | || |      | |   | |  \ \ / // _ \  
     | | | |\  || |_| || |_| || |___   | |   | |   \ V // ___ \ 
    |___||_| \_||____/  \___/  \____|  |_|  |___|   \_//_/   \_\
    
    To log in, you need an API key. You can obtain it from your account at https://console.inductiva.ai/account.
Please paste your API key here (input will not be visible): 
ERROR: Could not validate provided API Key.
```

**3. Logging Out:**
```
$ inductiva auth logout
Logout successful.
```

**4. Logging Out (Logged out:**
```
$ inductiva auth logout
Error: You are not logged in.
```

### CODE REVIEW

**Comment by Pbarbosa92** [comment]
Can't the login open a browser window to log in in the console and going that would create the file with the api key on the local computer ?

**Comment by rcvalerio** [comment]
> Can't the login open a browser window to log in in the console and going that would create the file with the api key on the local computer ?

It is possible to do it, however it might be too confusing for the user. Imagine a browser opens with the [account page](https://console.inductiva.ai/account). The user has no idea what he should do and most likely he will return to the terminal to read the instructions.

**Review by Pbarbosa92 — APPROVED** [review]

File: inductiva/_cli/cmd_login/__init__.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    prompt = getpass.getpass(
+        "    To log in, you need an API key. You can obtain it "
+        "from your account at https://console.inductiva.ai/account.\n"
+        "Please paste your API key here (input will not be visible): ")
+
+    api_key = prompt.strip()
+
+    utils.set_stored_api_key(api_key)
+    print("Login successful.")
```
[the remark this answers is not in the document]
Comment (rcvalerio) [reply inductiva/_cli/cmd_login/__init__.py]: I'm now calling `users.get_info()` to check the key and print a more informative message on login. I believe we don't have an endpoint that just checks the api key.

File: inductiva/_cli/cmd_login/__init__.py [hunk 2]
```diff
@@ -36,7 +36,15 @@ def login(_):
     api_key = prompt.strip()
 
     utils.set_stored_api_key(api_key)
-    print("Login successful.")
+
+    try:
+        user_info = users.get_info()
+    except Exception:
+        utils.remove_stored_api_key()
+        raise RuntimeError("Invalid API key. Please try again.")
```
[the remark this answers is not in the document]
Comment (rcvalerio) [reply inductiva/_cli/cmd_login/__init__.py]: I believe that was already changed. Maybe you reviewed old code?

**Pull request merged**